sonar-php-plugin/src/main/java/org/sonar/plugins/php/reports/phpstan/PhpStanRuleDefinition.java

```diff
@@ -19,21 +19,28 @@
  */
 package org.sonar.plugins.php.reports.phpstan;
 
-import org.sonar.api.server.rule.RulesDefinition;
-import org.sonar.plugins.php.api.Php;
+import javax.annotation.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.sonar.api.SonarRuntime;
+import org.sonar.plugins.php.reports.ExternalRulesDefinition;
 import org.sonarsource.analyzer.commons.ExternalRuleLoader;
 
 import static org.sonar.plugins.php.reports.phpstan.PhpStanSensor.PHPSTAN_REPORT_KEY;
 import static org.sonar.plugins.php.reports.phpstan.PhpStanSensor.PHPSTAN_REPORT_NAME;
 
-public class PhpStanRuleDefinition implements RulesDefinition {
+public class PhpStanRuleDefinition extends ExternalRulesDefinition {
+  private static final Logger LOG = LoggerFactory.getLogger(PhpStanRuleDefinition.class);
 
-  private static final String RULES_JSON = "org/sonar/plugins/php/reports/phpstan/rules.json";
-
-  static final ExternalRuleLoader RULE_LOADER = new ExternalRuleLoader(PHPSTAN_REPORT_KEY, PHPSTAN_REPORT_NAME, RULES_JSON, Php.KEY);
+  public PhpStanRuleDefinition(@Nullable SonarRuntime sonarRuntime) {
+    super(sonarRuntime, PHPSTAN_REPORT_KEY, PHPSTAN_REPORT_NAME);
+  }
 
-  @Override
-  public void define(Context context) {
-    RULE_LOADER.createExternalRuleRepository(context);
+  static ExternalRuleLoader ruleLoader() {
+    if (ruleLoader == null) {
+      LOG.debug("PHPStan importing not initialized at startup, initializing it now.");
+      ruleLoader = initializeRuleLoader(null, PHPSTAN_REPORT_KEY, PHPSTAN_REPORT_NAME);
+    }
+    return ruleLoader;
```

Comment:
I'm not a big fan of this logic, having a mutable static field `ruleLoader` that get's initialized during construction (in the super method).
But I tried a few different approaches, and I couldn't find a better one.
We're confined by DependencyInjection in the `PhpPlugin` class....
If you can think of a better solution, I would love to see it.

Reply:
I indeed find the logic a bit odd.
As both `PhpStanRuleDefinition` and `PsalmRulesDefinition` extend from the class `ExternalRulesDefinition`, the last object to be created override the static attribute `ruleLoader`.
It seems a bit dangerous to me as then the `define` method is using this `ruleLoader` to call `createExternalRuleRepository`.
Unless I misunderstood the structure it may/will create some issues.
I'm not sure if we have test cases to cover/check this, I only saw separate test class and separate ITs.

Reply:
Good catch!